### PR TITLE
Fix for some problems related to loops

### PR DESCRIPTION
**Pull request by andcarminati**
With this PR we remove 9 from 11 failures from QoR when using `-fno-unroll-loops`.

### CODE REVIEW

File: llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp [hunk 1]
```diff
@@ -485,9 +485,11 @@ int InterBlockScheduling::getCyclesToRespectTiming(
     for (auto &Bundle : R.Bundles) {
       for (MachineInstr *MI : Bundle.getInstrs()) {
         DistancesFromLoopEntry[MI] = DistFromLoopEntry;
-        Edges.addNode(MI);
       }
-      ++DistFromLoopEntry;
```
Comment (konstantinschwarz) [inline llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp:490]: curious: any reason to change this from pre-increment to post-increment?

Comment (andcarminati) [reply llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp:490]: Hi, good eyes. Probably I deleted this increment and retyped it again, ! I will change it to be the same as before.

File: llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp [hunk 2]
```diff
@@ -485,9 +485,11 @@ int InterBlockScheduling::getCyclesToRespectTiming(
...
+      DistFromLoopEntry++;
+    }
+    for (auto MI = R.begin(); MI != R.end(); MI++) {
```
Comment (gbossu) [inline llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp]: Nit: Add a comment to mention that this is iterating in semantic order, not scheduled order.
Nit2: Can we use a loop like `for (const MachineInstr* MI : R) { ... }` instead?

**Review by gbossu — DISMISSED** [review]
LGTM, great job finding those bugs!

File: llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp [hunk 3]
```diff
@@ -726,7 +726,8 @@ std::optional<bool> ZeroOverheadLoop::createTripCountGreaterCondition(
 
 void ZeroOverheadLoop::adjustTripCount(int TripCountAdjust) {
   LLVM_DEBUG(dbgs() << "TripCountAdjust =  " << TripCountAdjust << "\n");
-  if (DefTripCount->getOperand(1).isImm()) {
+  if (DefTripCount->getOperand(1).isImm() &&
```
Comment (andcarminati) [inline llvm/lib/Target/AIE/AIEBasePipelinerLoopInfo.cpp:729]: This conflicts with https://github.com/Xilinx/llvm-aie/pull/146. As agreed with @martien-de-jong  his solution will cover this case, so this is just a temporary fix.

**Review by gbossu — APPROVED** [review]

**Pull request merged**